avatars/healthControl.js

```diff
@@ -0,0 +1,36 @@
+
+import gameManager from '../game';
+import metaversefile from 'metaversefile';
+export class HealthControl {
+  constructor(isLocalPlayer) {
+    this.health = 100;
+    this.mana = 100;
+    this.isLocalPlayer = isLocalPlayer;
+    window.addEventListener('DOMContentLoaded', () => {
+      this.display();
+    });
+  }
+
+  takeDamage(damage) {
+    this.health -= damage;
+    if (this.health <= 0) {
+      this.health = 0;
+      console.log('player is incapacitated')
+      // play death animation etc
+    }
+    // gameManager.playerDiorama.toggleShader();
+    // window.dispatchEvent(new MessageEvent('playerHit'))
+    this.display();
+  }
+
+  display() {
+    if (this.isLocalPlayer) {
+    document.getElementById('healthDisplay').innerHTML = `HP: ${this.health}`;
```

Comment:
There is a better way to do this. We generally don't want to access deep DOM like this, because the engine is imported in many other contexts besides this, which might not have this DOM, or even any DOM.

Therefore, we do bindings explicitly and there is an entire bootstrap procedure dedicated to that:
https://github.com/webaverse/app/blob/1e6b31cab9efa3e23941fcde552fff145cdbadc4/src/App.jsx#L22
https://github.com/webaverse/app/blob/277711fa98e160496af609bd1200876c2c22db61/webaverse.js#L134

This kind of canvas should be injected through that procedure. Maybe this is the character system's `bindCanvas()` method.

Structuring the code that way ensures that environment dependencies have a clear point of control, rather than being sprinkled around deeply in the code like this. It makes it much easier to reason about and test the code.